js/coinsbit.js

```diff
@@ -0,0 +1,520 @@
+        const request = {
+            'market': market['id'],
+            'orderId': parseInt (id),
+        };
+        const response = await this.privatePostOrderCancel (this.extend (request, params));
+        const result = this.safeValue (response, 'result');
+        return this.parseOrder (result, market);
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        const orders = await this.fetchOpenOrders (symbol);
+        const targetOder = this.findOrderById (id, orders);
+        if (targetOder !== undefined) {
+            return targetOder;
+        } else {
+            throw new OrderNotFound ('\nid: ' + id);
+        }
+    }
+
+    findOrderById (id, orders) {
+        for (let orderIndex = 0; orderIndex < orders.length; orderIndex++) {
+            if (orders[orderIndex]['id'] === id) {
+                return orders[orderIndex];
+            }
+        }
+        return undefined;
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        return await this.fetchOpenOrders (symbol, since, limit, params);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + this.version + '/';
+        if (api === 'public') {
+            url += 'public/';
+        }
+        url += this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        if (api === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const request = '/api/' + this.version + '/' + this.implodeParams (path, params);
+            query = this.extend ({
+                'request': request,
+                'nonce': this.nonce ().toString (),
+            }, query);
+            body = this.json (query);
+            const payload = this.stringToBase64 (body);
+            const secret = this.encode (this.secret);
+            const signature = this.hmac (payload, secret, 'sha512');
+            headers = {
+                'Content-type': 'application/json',
+                'X-TXC-APIKEY': this.apiKey,
+                'X-TXC-PAYLOAD': payload,
+                'X-TXC-SIGNATURE': signature,
+            };
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        if (response === undefined) {
+            return;
+        }
+        if (code !== 200) {
+            const feedback = '\nid: ' + this.id + '\nurl: ' + url + '\ncode: ' + code + '\nbody:\n' + body;
+            this.throwExactlyMatchedException (this.httpExceptions, code.toString (), feedback);
+        }
+        if (body.length > 0) {
+            if (body[0] === '{') {
```

Comment:
if `response === undefined` these conditionals will always fail, `response` is always JSON or undefined. Also it violates the CONTRIBUTING guidelines here:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#working-with-array-lengths

So i suggest to remove these lines:

```JavaScript
        if (body.length > 0) {
            if (body[0] === '{') {
```